File: PB - More Exercises Python/3.0 Nested Conditional Statements - More Exercises/08. Point on Rectangle Border.py
x1, y1, x2, y2, x, y = [float(input()) for _ in range(6)]

if (x == x1 and y1 <= y <= y2) or (x == x2 and y1 <= y <= y2) or \
    (y == y1 and x1 <= x <= x2) or (y == y2 and x1 <= x <= x2):
    print("Border")
elif y == y1 and y == y2 and x1 < x < x2:
    print("Border")
else:
    print("Inside / Outside")


'''
2
-3
12
3
8
-1
'''

File: PB - More Exercises GO/3.0 Nested Conditional Statements - More Exercises/08. Point on Rectangle Border.go
package main

import "fmt"

func main() {
	var x1, y1, x2, y2, x, y float64
	fmt.Scan(&x1, &y1, &x2, &y2, &x, &y)

	if (x == x1 && y1 <= y && y <= y2) || (x == x2 && y1 <= y && y <= y2) ||
		(y == y1 && x1 <= x && x <= x2) || (y == y2 && x1 <= x && x <= x2) {
		fmt.Println("Border")
	} else if y == y1 && y == y2 && x1 < x && x < x2 {
		fmt.Println("Border")
	} else {
		fmt.Println("Inside / Outside")
	}

}

/*
name   :08. Point on Rectangle Border
input  :2 -3 12 3 1 3
output :Inside / Outside
*/
